src/CSS.cpp

```diff
@@ -13,36 +13,49 @@ namespace CSS
     {
         sProvider = gtk_css_provider_new();
 
-        struct CSSDir
+        std::vector<std::string> locations;
+        std::string home = std::getenv("HOME");
+        const char* env;
+
+        if ((env = std::getenv("XDG_CONFIG_HOME")))
+        {
+            locations.push_back(env);
+        }
+        else if (!home.empty())
+        {
+            locations.push_back(home.append("/.config"));
+        }
+
+        if ((env = std::getenv("XDG_DATA_HOME")))
+        {
+            locations.push_back(env);
+        }
+        else if (!home.empty())
         {
-            std::string xdgEnv;
-            std::string fallbackPath;
-            std::string relPath;
-        };
-        std::string home = getenv("HOME");
-        std::array<CSSDir, 4> locations = {
-            CSSDir{"XDG_CONFIG_HOME", home + "/.config",      "/gBar/style.css"}, // Local config
-            CSSDir{"XDG_DATA_HOME",   home + "/.local/share", "/gBar/style.css"}, // local user install
-            CSSDir{"",                "/usr/local/share",     "/gBar/style.css"}, // local all install
-            CSSDir{"",                "/usr/share",           "/gBar/style.css"}, // package manager all install
-        };
+            locations.push_back(home.append("/.local/share"));
+        }
+
+        if ((env = std::getenv("XDG_DATA_DIRS")))
+        {
+            std::stringstream ss(env);
+            std::string dir;
+            while (std::getline(ss, dir, ':'))
+                locations.push_back(dir);
+        }
+        else
+        {
+            locations.push_back("/usr/local/share");
+            locations.push_back("/usr/share");
+        }
 
         GError* err = nullptr;
         for (auto& dir : locations)
         {
-            const char* xdgConfig = dir.xdgEnv.size() ? getenv(dir.xdgEnv.c_str()) : nullptr;
-            std::string file;
-            if (xdgConfig)
-            {
-                file = (std::string(xdgConfig) + dir.relPath).c_str();
-            }
-            else
-            {
-                file = (dir.fallbackPath + dir.relPath).c_str();
-            }
+            std::string file = dir.append("/gBar/style.css");
```

Comment:
While we're at it with the ```append```'s:
```
dir + "/gBar/style.css"
```
```append``` causes an unnecessary resize here, while "+" doesn't. (Functionally it is the same though :smile: )

Reply:
Will change it anyways :)